### DISCUSSION TITLE
Is there a clean way to implement refresh tokens?

### DISCUSSION THREAD

**Discussion by brandonpittman** [sergiodxa/remix-auth#215] [Q&A] [answered]
I have a hand-rolled auth system that does refresh tokens like [this article](https://sergiodxa.com/articles/working-with-refresh-tokens-in-remix) @sergiodxa wrote. Is there a way to cleanly do this with remix-auth, or do I still need a custom authenticate function?

**Comment by sergiodxa** [comment, marked as answer]
Instead of `let accessToken = session.get("accessToken");` replace it with `isAuthenticate` to get the accessToken and do the rest in the same way.

**Reply by brandonpittman** [reply]
Had a feeling that was the best path. Thanks.

**Reply by nathankirk91** [reply]
Thank you, this helped me a lot!
